DeviceDiagnostics/DeviceDiagnostics.cpp

```diff
@@ -241,12 +238,10 @@ namespace WPEFramework
                 return;
             }
 
+            /* add/update map with new state and notify any listener */
             (*decoderStatus)[std::string(eventData->pipeline_id)] = decoderInfo;
-
-            /* this pipeline has been most recently read, send event
-             * to inform caller status has changed */
-            if (strcmp(eventData->pipeline_id, lastDecoder->c_str()) == 0)
-                t->onDecoderStatusChange(eventData->decoder, eventData->status);
+            t->onDecoderStatusChange(eventData->decoder, eventData->status);
```

Comment:
The onVideoDecoderStatusChanged and onAudioDecoderStatusChanged events should also aggregate (call getMostActiveDecoderStatus) and then fire the event if the aggregate has changed.